pkg/processor/trigger/nats/test/nats_test.go

```diff
@@ -0,0 +1,138 @@
+package test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/dockerclient"
+	"github.com/nuclio/nuclio/pkg/errors"
+	"github.com/nuclio/nuclio/pkg/platform"
+	"github.com/nuclio/nuclio/pkg/processor/test/suite"
+
+	"github.com/nats-io/go-nats"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	natsPort  = 4222
+	topicName = "my.topic"
+)
+
+type NatsTestSuite struct {
+	processorsuite.TestSuite
+	natsContainerId  string
+	natsSubscription *nats.Subscription
+	natsPort         int
+	topicName        string
+}
+
+func (suite *NatsTestSuite) SetupSuite() {
+	var err error
+
+	suite.TestSuite.SetupSuite()
+
+	suite.natsPort = natsPort
+	suite.topicName = topicName
+
+	// Start nats
+	suite.natsContainerId, err = suite.DockerClient.RunContainer("nats:latest",
+		&dockerclient.RunOptions{
+			Ports: map[int]int{suite.natsPort: suite.natsPort, 8222: 8222},
+		})
+
+	suite.Require().NoError(err, "Failed to start NATS container")
+}
+
+func (suite *NatsTestSuite) SetupTest() {
+	suite.TestSuite.SetupTest()
+
+	suite.FunctionDir = suite.getFunctionsPath()
+}
+
+func (suite *NatsTestSuite) TearDownSuite() {
+	suite.TestSuite.TearDownTest()
+
+	// If we weren't successful starting, nothing to do
+	if suite.natsContainerId != "" {
+		suite.DockerClient.RemoveContainer(suite.natsContainerId)
+	}
+}
+
+func (suite *NatsTestSuite) TestPostEventPython() {
+	suite.invokeEventRecorder("event_recorder_python", "python")
+}
+
+func (suite *NatsTestSuite) TestPostEventGolang() {
+	suite.invokeEventRecorder(path.Join("_event_recorder_golang", "event_recorder.go"), "golang")
+}
+
+func (suite *NatsTestSuite) invokeEventRecorder(functionPath string, runtimeType string) {
+	suite.Runtime = runtimeType
+
+	deployOptions := suite.GetDeployOptions("event_recorder",
+		suite.GetFunctionPath(functionPath))
+
+	suite.DeployFunction(deployOptions, func(deployResult *platform.DeployResult) bool {
+
+		natsCon, err := nats.Connect(nats.DefaultURL)
+		if err != nil {
+			errors.Wrapf(err, "Can't connect to NATS server %s", nats.DefaultURL)
+			return false
+		}
+
+		// Send 3 messages
+		for requestIdx := 0; requestIdx < 3; requestIdx++ {
+			natsCon.Publish(suite.topicName, []byte(fmt.Sprintf(`{"request": "%d"}`, requestIdx)))
```

Comment:
Always, always, always check errors. I have yet to enable error checking in the linting so please pay special attention not to miss error checking.